src/pool-bin/libraries/BinPool.sol

```diff
@@ -168,6 +169,14 @@ library BinPool {
                         amountsInWithFees = amountsInWithFees.sub(pFee);
                     }
 
+                    if (
```

Comment:
can i trouble you to try to add some test cases for swap/mint case? would be good exercise to further our knowledge on bin related especially with https://github.com/pancakeswap/pancake-v4-core/pull/213 coming